src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponentSpec.scala

```diff
@@ -0,0 +1,177 @@
+    val entityId2 = runAndWait((entityQuery returning entityQuery.map(_.id)) += EntityRecord(0, "name2", "type", workspaceId))
+    val entityId3 = runAndWait((entityQuery returning entityQuery.map(_.id)) += EntityRecord(0, "name3", "type", workspaceId))
+    val entityId4 = runAndWait((entityQuery returning entityQuery.map(_.id)) += EntityRecord(0, "name4", "type", workspaceId))
+
+    val testAttribute = AttributeEntityReferenceList(Seq(AttributeEntityReference("type", "name1"), AttributeEntityReference("type", "name2"), AttributeEntityReference("type", "name3"), AttributeEntityReference("type", "name4")))
+    val insertedIds = attributeQuery.insertAttributeRecords("test", testAttribute, workspaceId).map(x => runAndWait(x))
+    assertResult(4) { insertedIds.size }
+
+    assertResult(Set(
+      AttributeRecord(0, "test", None, None, None, Option(entityId1), Option(0)),
+      AttributeRecord(0, "test", None, None, None, Option(entityId2), Option(1)),
+      AttributeRecord(0, "test", None, None, None, Option(entityId3), Option(2)),
+      AttributeRecord(0, "test", None, None, None, Option(entityId4), Option(3)))) {
+
+      runAndWait(attributeQuery.filter(_.id inSet insertedIds).result).map(_.copy(id=0)).toSet
+    }
+  }
+
+  it should "throw exception inserting ref to nonexistent entity" in {
+    val workspaceId = UUID.randomUUID()
+    runAndWait(workspaceQuery += WorkspaceRecord("testns", "testname1", workspaceId, "bucket", new Timestamp(0), new Timestamp(0), "me", false))
+    val testAttribute = AttributeEntityReference("type", "name")
+    intercept[RawlsException] {
+      attributeQuery.insertAttributeRecords("test", testAttribute, workspaceId).map(x => runAndWait(x))
+    }
+  }
+
+  it should "throw exception inserting inconsistent list values" in {
+    val testAttribute = AttributeValueList(Seq(AttributeNumber(9), AttributeString("oops"), AttributeNumber(7), AttributeNumber(6)))
+    intercept[RawlsException] {
+      attributeQuery.insertAttributeRecords("test", testAttribute, UUID.randomUUID()).map(x => runAndWait(x))
+    }
+  }
+
+  it should "delete attribute records" in {
+    val inserts = Seq(
+      (attributeQuery returning attributeQuery.map(_.id)) += AttributeRecord(0, "test1", None, Some(1), None, None, None),
+      (attributeQuery returning attributeQuery.map(_.id)) += AttributeRecord(0, "test2", None, Some(2), None, None, None),
+      (attributeQuery returning attributeQuery.map(_.id)) += AttributeRecord(0, "test3", None, Some(3), None, None, None)) map { insert =>
+
+      runAndWait(insert)
+    }
+    val attributeRecs = runAndWait(attributeQuery.filter(_.id inSet inserts).result)
+    assertResult(3) { attributeRecs.size }
+
+    assertResult(3) { runAndWait(attributeQuery.deleteAttributeRecords(attributeRecs)) }
+
+    assertResult(0) { runAndWait(attributeQuery.filter(_.id inSet inserts).result).size }
+  }
+
+  it should "unmarshall attribute records" in {
```

Comment:
Add a test for the case where some, but not all, members of a list have None for the index